lib/practice_exercises.rb

```diff
@@ -1,13 +1,44 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) - as string increases in length, time increases
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  split_string = string.downcase.gsub(/[\,\s:]/, '')
+  first = 0
+  last = (split_string.length - 1)
+  
+  if split_string.empty? || split_string.length == 1
+    return true
+  end
+  
+  while first < last
+    if split_string[first] != split_string[last]
+      return false
+    else 
+      first += 1
+      last -= 1
+    end
+    
+    return true
+  end
+  
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+
+# Time Complexity: O(n) - n is the length of the shortest word in the array
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍